pkg/tls/certificatemanagement/controller/ca.go

```diff
@@ -0,0 +1,533 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
```

Comment:
I have a similar comment to what I said about the render subpackage, I don't think that this should be in a controller subpackage (for the same reasons I said you shouldn't have a render subpackage here).

I think you we should draw a better line between what's controller specific logic that should be in the controller package, what is ca logic that should be in this package, and what (if anything) should be in the render package.

For instance, you could say the whole of CertificateManager implementation and interface is a product of the controller, that only needs to be used by the controller, so having that interface and implementation in a subpackage in the controller package makes sense. You could also very easily abstract out the getting the imageset / setting the status into functions that live in the contrller pacakage, i.e. `ImageSetForCertManager(installation)` / `SetStatusForCertManagement(cli, certmanager)`.

The KeyCert and Certificate interfaces / implementations need to be used in multiple different packages, including render and controller, so having the definition here (under the controller subpackage) doesn't seem to make the best sense.